hamilton/plugins/h_ray.py

```diff
@@ -105,25 +111,36 @@ def __init__(self, result_builder: base.ResultMixin):
             )
 
     @staticmethod
-    def check_input_type(node_type: typing.Type, input_value: typing.Any) -> bool:
+    def do_validate_input(node_type: typing.Type, input_value: typing.Any) -> bool:
         # NOTE: the type of a raylet is unknown until they are computed
         if isinstance(input_value, ray._raylet.ObjectRef):
             return True
         return htypes.check_input_type(node_type, input_value)
 
     @staticmethod
-    def check_node_type_equivalence(node_type: typing.Type, input_type: typing.Type) -> bool:
-        return node_type == input_type
-
-    def execute_node(self, node: node.Node, kwargs: typing.Dict[str, typing.Any]) -> typing.Any:
+    def do_check_edge_types_match(type_from: typing.Type, type_to: typing.Type) -> bool:
+        return type_from == type_to
+
+    # def execute_node(self, node: node.Node, kwargs: typing.Dict[str, typing.Any]) -> typing.Any:
+    #     """Function that is called as we walk the graph to determine how to execute a hamilton function.
+
+    #     :param node: the node from the graph.
+    #     :param kwargs: the arguments that should be passed to it.
+    #     :return: returns a ray object reference.
+    #     """
+    #     ray_options = parse_ray_remote_options_from_tags(node.tags)
+    #     return ray.remote(raify(node.callable)).options(**ray_options).remote(**kwargs)
+    
+    def do_remote_execute(self, *, execute_lifecycle_for_node : typing.Callable, node: node.Node, kwargs: typing.Dict[str, typing.Any],**future_kwargs: typing.Any) -> typing.Any:
         """Function that is called as we walk the graph to determine how to execute a hamilton function.
 
-        :param node: the node from the graph.
+        :param execute_lifecycle_for_node: wrapper function that executes lifecycle hooks and methods
         :param kwargs: the arguments that should be passed to it.
         :return: returns a ray object reference.
         """
         ray_options = parse_ray_remote_options_from_tags(node.tags)
-        return ray.remote(raify(node.callable)).options(**ray_options).remote(**kwargs)
+        return ray.remote(raify(execute_lifecycle_for_node))#.options(**ray_options).remote()#**kwargs)
```

Comment:
So here we want to:

1. raify `execute_lifecycle_for_node` -- this will inject `ray.remote` around that function.
2. pass in options from node tags
3. then  **kwargs here should include all the arguments to execute `execute_lifecycle_for_node`

Reply:
So the trickiest part here is connecting the arguments required for the new function, as well as the arguments required for the actual node function call itself.